src/player.rs

```diff
@@ -113,39 +115,40 @@ fn setup(
 				},
 				..default()
 			},
-			TextureAtlas {
+			texture_atlas: TextureAtlas {
 				layout: atlas_handle,
 				index: PLAYER_CONFIG.sprite_idx_stand,
 			},
-		))
-		.insert(RigidBody::KinematicPositionBased)
-		.insert(Collider::cuboid(
-			PLAYER_CONFIG.sprite_render_width / 2.0,
-			PLAYER_CONFIG.sprite_render_height / 2.0,
-		))
-		.insert(KinematicCharacterController::default())
-		.insert(Direction::Right)
-		.insert(Animation::new(
-			PLAYER_CONFIG.sprite_idx_idle,
-			PLAYER_CONFIG.cycle_delay,
-		));
+			animation: Animation::new(
+				PLAYER_CONFIG.sprite_idx_idle,
+				PLAYER_CONFIG.cycle_delay,
+			),
+			velocity: Velocity {
+				x: PLAYER_CONFIG.player_velocity_x,
+				y: PLAYER_CONFIG.player_velocity_y,
+			},
+			..default()
+		},
+	));
 }
 
 fn movement(
 	input: Res<ButtonInput<KeyCode>>,
 	time: Res<Time>,
-	mut query: Query<&mut KinematicCharacterController>,
+	mut player_query: Query<&mut KinematicCharacterController>,
+	velocity_query: Query<&Velocity, With<Player>>,
```

Comment:
I turned velocity in a component to be able to query for it. My thinking here is that many different entities _could_ have velocity, and we should be able to query for it to update or read. Ex. if we wanted to add a mechanism that slows down all enemies or makes the , this would be how we could do that